clients/include/testing_tbmv_batched.hpp

```diff
@@ -0,0 +1,245 @@
+    if(!dA || !dx)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_tbmv_batched<T>(
+            handle, uplo, transA, diag, M, K, nullptr, lda, dx, incx, batch_count),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_tbmv_batched<T>(
+            handle, uplo, transA, diag, M, K, dA, lda, nullptr, incx, batch_count),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_tbmv_batched<T>(nullptr, uplo, transA, diag, M, K, dA, lda, dx, incx, batch_count),
+        rocblas_status_invalid_handle);
+}
+
+template <typename T>
+void testing_tbmv_batched(const Arguments& arg)
+{
+    rocblas_int       M           = arg.M;
+    rocblas_int       K           = arg.K;
+    rocblas_int       lda         = arg.lda;
+    rocblas_int       incx        = arg.incx;
+    char              char_uplo   = arg.uplo;
+    char              char_diag   = arg.diag;
+    rocblas_fill      uplo        = char2rocblas_fill(char_uplo);
+    rocblas_operation transA      = char2rocblas_operation(arg.transA);
+    rocblas_diagonal  diag        = char2rocblas_diagonal(char_diag);
+    rocblas_int       batch_count = arg.batch_count;
+
+    rocblas_local_handle handle;
+
+    // argument sanity check before allocating invalid memory
+    if(M < 0 || K < 0 || lda < M || lda < 1 || !incx || K >= lda || batch_count <= 0)
+    {
+        static const size_t     safe_size = 100; // arbitrarily set to 100
+        device_vector<T*, 0, T> dA1(safe_size);
+        device_vector<T*, 0, T> dx1(safe_size);
+        if(!dA1 || !dx1)
+        {
+            CHECK_HIP_ERROR(hipErrorOutOfMemory);
+            return;
+        }
+
+        EXPECT_ROCBLAS_STATUS(
+            rocblas_tbmv_batched<T>(
+                handle, uplo, transA, diag, M, K, dA1, lda, dx1, incx, batch_count),
+            (M < 0 || K < 0 || lda < M || lda < 1 || !incx || K >= lda || batch_count < 0)
+                ? rocblas_status_invalid_size
+                : rocblas_status_success);
+
+        return;
+    }
+
+    device_vector<T*, 0, T> dA(batch_count);
+    device_vector<T*, 0, T> dx(batch_count);
+
+    if(!dA || !dx)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    size_t size_A = lda * size_t(M);
+    size_t size_x, abs_incx;
+
+    abs_incx = incx >= 0 ? incx : -incx;
+    size_x   = M * abs_incx;
+
+    // Naming: dK is in GPU (device) memory. hK is in CPU (host) memory
+    host_vector<T> hA[batch_count];
+    host_vector<T> hx[batch_count];
+    host_vector<T> hx_1[batch_count];
+    host_vector<T> hx_gold[batch_count];
+
+    device_batch_vector<T> bA(batch_count, size_A);
+    device_batch_vector<T> bx(batch_count, size_x);
+
+    int last = batch_count - 1;
+    if((!bA[last] && size_A) || (!bx[last] && size_x))
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
```

Comment:
No.
use CHECK_HIP_ERROR(bA.memcheck()) and CHECK_HIP_ERROR(bx.memcheck())

Reply:
I think you're looking at an old commit, correct me if I'm wrong but I think this was fixed in 
33bf010.

Reply:
you are right Daine, my mistake.